src/btree/bt_read.c

```diff
@@ -248,11 +249,26 @@ __page_read(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags)
         deltas = NULL;
 
     /* Build a full disk image of the page after reading from disk. */
-    if (/* DISABLES CODE */ (0) && count > 1) {
+    //if (/* DISABLES CODE */ (0) && count > 1) {
+    // if ( ref->page != NULL && F_ISSET(ref, WT_REF_FLAG_INTERNAL) && count > 1) {
+    if (F_ISSET(ref, WT_REF_FLAG_INTERNAL) && count > 1) {
+        size_t new_image_buf_size;
+        uint32_t split_size;
+        WT_BTREE *btree;
+
+        btree = S2BT(session);
+
+        /* Allocate enough size for the new image, similar to __rec_split_chunk_init. */
+        split_size = __wt_split_page_size(btree->split_pct, btree->maxleafpage, btree->allocsize);
+        new_image_buf_size = 2 * WT_ALIGN(WT_MAX(btree->maxleafpage, split_size), btree->allocsize);
+        WT_ERR(__wt_scr_alloc(session, 0, &new_image));
+        WT_ERR(__wt_buf_init(session, new_image, new_image_buf_size));
```

Comment:
Why this code is moved here? Though we allocate a large buffer, it may still not be enough and we need to handle that.